src/hooks/UrqlUseQuery.re

```diff
@@ -24,12 +26,17 @@ type useQueryState('a) = {
 }
 type useQueryResponse('a) = (useQueryState('a), partialOperationContextFn);
 
-let useQueryResponseToRecord = (result: useQueryStateJs('a)) => {
-  let data = result->dataGet;
+let useQueryResponseToRecord =
+    (parse: Js.Json.t => 'response, result: useQueryStateJs('a)) => {
+  let data =
+    switch (result->dataGet) {
+    | None => None
+    | Some(d) => Some(parse(d))
```

Comment:
I believe that it's now safe to remove the type parameter from `useQueryStateJs` and `useQueryResponseJs`, as the only type `option('a)` that satisfies this switch is `option(Js.Json.t)`, as the only type accepted by `parse` is `Js.Json.t`

This switch could also become: `result->dataGet->Belt.Option.map(parse)`, but I don't know about the thoughts of using too much piping in this project, as it can be a little alien to newcomers.

Reply:
Good point @gugahoa that type parameter is unnecessary now. @Schmavery do you mind implementing this before we merge? I'm cool with piping and `Belt.Option.map`, I think it feels a little nicer than this `switch` if a little intimidating at first for newcomers.